lib/api/client.js

```diff
@@ -125,6 +125,11 @@ Client.prototype.makeRequest = function(obj) {
 			callback(null, resData); //no error, so no err object
 		});
 	});
+	
+	
+	req.on('error', (e) => {
```

Comment:
This seems like a good idea, lets do it similar to how the timeout is handled

```js
	req.on('error', error => {
		errorHandler.handle({ reqError: true, error }, '', callback, self.options.get('format'));
		req.destroy();
	});
```

and in `error-handler.js`

```js
	if (res.reqError) {
		err = {
			error: true,
			errors: [res.error],
			statusCode: 'request-error',
			text: ''
		};
		callback(err, res);
		return;
	}
```

Also, please add tests to `client.spec.js` and `error-handler.spec.js`